Dfe.Data.SearchPrototype/Infrastructure/Tests/CognitiveSearchServiceAdapterTests.cs

```diff
@@ -49,14 +50,18 @@ public async Task Search_With_Valid_SearchContext_Returns_Configured_Results()
     }
 
     [Fact]
-    public Task Search_With_Valid_SearchContext_No_Options_Returned_Throws_ApplicationException()
+    public Task Search_WithNoSearchOptions_ThrowsApplicationException()
     {
+        var mockService = SearchServiceTestDouble.MockSearchService("SearchKeyword", "TargetCollection");
+        var mockSearchOptionsFactory = SearchOptionsFactoryTestDouble.MockForNoOptions();
+        var mockMapper = AzureSearchResponseToSearchResultsMapperTestDouble.MockMapperReturningEmptyResults();
+
         // arrange
         ISearchServiceAdapter cognitiveSearchServiceAdapter =
             CreateServiceAdapterWith(
-                SearchServiceTestDouble.MockSearchService("SearchKeyword", "TargetCollection"),
-                SearchOptionsFactoryTestDouble.MockForDefaultResult(),
-                AzureSearchResponseToSearchResultsMapperTestDouble.MockDefaultMapper());
+                mockService,
+                mockSearchOptionsFactory,
+                mockMapper);
```

Comment:
Possibly a comment for a different PR (so feel free to disregard / resolve this comment without changes), but I would be interested to understand the rationale for having an additional `CreateServiceAdapterWith` static method, as opposed to in-lining it (and removing a level of indirection) like this:

```csharp
        // arrange
        ISearchServiceAdapter cognitiveSearchServiceAdapter =
            new CognitiveSearchServiceAdapter<Establishment>(mockService, mockSearchOptionsFactory, mockMapper);
```

Could just be a historical thing where an older version was more complex, perhaps? Or just a style thing? Or to help with the generics side of things maybe?

_(regardless, for me this is just a curiosity and possibly style observation -- certainly nothing I want to get hung up on or block the PR for! 👍)_

Reply:
Don't know about why it's there, so I kept as is